### PR TITLE
improved gas for _process func

### PR DESCRIPTION
**Pull request by fritzschoff**

### CODE REVIEW

File: protocol/oracle-manager/contracts/modules/NodeModule.sol [hunk 1]
```diff
@@ -128,19 +128,26 @@ contract NodeModule is INodeModule {
     function _process(bytes32 nodeId) internal view returns (NodeOutput.Data memory price) {
         NodeDefinition.Data storage nodeDefinition = NodeDefinition.load(nodeId);
 
-        // Retrieve the output of the node's parents
-        NodeOutput.Data[] memory parentNodeOutputs = new NodeOutput.Data[](
-            nodeDefinition.parents.length
-        );
-        for (uint256 i = 0; i < nodeDefinition.parents.length; i++) {
-            parentNodeOutputs[i] = this.process(nodeDefinition.parents[i]);
-        }
```
[the remark this answers is not in the document]
Comment (fritzschoff) [reply protocol/oracle-manager/contracts/modules/NodeModule.sol:137]: with a var that is outside of the if statement that is potential undefined? 
Guess this still costs gas to declare a var
If I understood you correctly

[the remark this answers is not in the document]
Comment (fritzschoff) [reply protocol/oracle-manager/contracts/modules/NodeModule.sol:137]: so we are doing code beauty vs. gas consumption. I would agree with you if this code is javascript.

[the remark this answers is not in the document]
Comment (fritzschoff) [reply protocol/oracle-manager/contracts/modules/NodeModule.sol:137]: sounds good

[the remark this answers is not in the document]
Comment (ajsantander) [reply protocol/oracle-manager/contracts/modules/NodeModule.sol:137]: IMO you are both right. Solidity tends to make one prioritize gas optimization over code beauty in cases of functions that get hit a lot. However, code duplication is something we must avoid but in the most extreme cases, e.g. some ultra efficient math library. What I would recommend here is abstracting the duplicated logic into an internal or private function, both achieving better performance and deduping.

[the remark this answers is not in the document]
Comment (fritzschoff) [reply protocol/oracle-manager/contracts/modules/NodeModule.sol:137]: thanks. Will write a pure func

**Review by ajsantander — APPROVED** [review]

**Pull request merged**